### ISSUE TITLE
crash on latest version of flutter in master channel

### ISSUE THREAD

**Issue by DJafari** [tekartik/sqflite#233]
after upgrade flutter to `1.7.10-pre.19` my app crashed on startup with this message :

```
E/AndroidRuntime( 5859): FATAL EXCEPTION: Sqflite
E/AndroidRuntime( 5859): Process: *, PID: 5859
E/AndroidRuntime( 5859): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: Sqflite
E/AndroidRuntime( 5859): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:794)
E/AndroidRuntime( 5859): 	at io.flutter.embedding.engine.FlutterJNI.invokePlatformMessageResponseCallback(FlutterJNI.java:727)
E/AndroidRuntime( 5859): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:140)
E/AndroidRuntime( 5859): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:225)
E/AndroidRuntime( 5859): 	at com.tekartik.sqflite.SqflitePlugin$6.run(SqflitePlugin.java:778)
E/AndroidRuntime( 5859): 	at android.os.Handler.handleCallback(Handler.java:873)
E/AndroidRuntime( 5859): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/Androiduntime( 5859): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime( 5859): 	at android.os.HandlerThread.run(HandlerThread.java:65)
I/Process ( 5859): Sending signal. PID: 5859 SIG: 9
```

**Comment by tamhiupanTam** [comment]
This is because there is an update in flutter dev/master branch which do not allow FlutterJNI to run on non-main-thread (https://github.com/flutter/engine/commit/2c9e37c) whereas the sqflite 1.1.6 changed to run the sqf database on background thread in Android. 
As of now, I think there is two option to resolve this,
1. Go back to flutter stable channel
2. Change the sqflite version back to 1.1.5

**Comment by DJafari** [comment]
@tamhiupanTam thanks for your help

i switched flutter to `1.5.8` version to solved this crash, but my goal was to report this problem to resolved

**Comment by alextekartik** [comment]
Thanks for the report. I think even sqflite 1.1.5 should crash since many handlers do run and send result in a background thread. I was not aware of this incoming change. I think the solution will be to ensure that all  results are sent from the UI thread (they can still run in a background thread). That should not be an issue. I don't know if iOS has the same restriction yet.

**Comment by alextekartik** [comment]
It should be fixed on 1.1.6+1 just published. That was a simple fix. Let me know if that works for you and thanks again for the report, it is impossible for me do to extensive testing on every master build (until I setup a CI where I can run native tests)
